oereb_gewaesserschutz/build.gradle

```
@@ -179,7 +190,28 @@ task importDataToLive(type: Ili2pgReplace) {
     dbschema = "live"
     dataFile = file(Paths.get("$rootDir/" + xtfFileName))
     dataset = groundwaterProtectionDataSet
-    disableValidation = true 
+    disableValidation = true
+    
+}
+
+task zipXtfFile(type: Zip) {
+    description = "Zipt das xtf-File mit den Gewässerschutzdaten für den Upload nach S3"
+    from "$rootDir/"
+    include xtfFileName
+    archiveName zipFileName
+    destinationDir(file("$rootDir"))
+}
+
+task uploadXtfToS3Geodata(type: S3Upload, dependsOn: "zipXtfFile") {
+    description = "Xtf-File nach S3 hochladen."
+    accessKey = awsAccessKeyAgi
+    secretKey = awsSecretAccessKeyAgi
+    //sourceDir = file("$rootDir/")
+    sourceFile = file(Paths.get("$rootDir/" + zipFileName))
```

Ginge hier nicht auch 
```suggestion
    sourceFile = file(Paths.get($rootDir, zipFileName))
```
(`Paths.get` sollte die Bestandteile eines Pfades selber sauber zusammensetzen können.)

---

Die "" brauchts schon, sonst kommt folgender Fehler:
Could not get unknown property '$rootDir' for task ':uploadXtfToS3Geodata' of type ch.so.agi.gretl.tasks.S3Upload.

---

Ermittlung, wie der s3TargetBucket je nach Umgebung heisst, auf eine Zeile vereinfacht. Das muss man nicht genau verstehen???

---

> Die "" brauchts schon, sonst kommt folgender Fehler:
> Could not get unknown property '$rootDir' for task ':uploadXtfToS3Geodata' of type ch.so.agi.gretl.tasks.S3Upload.

Ah OK, dann also vielleicht so (d.h. einfach ohne Slash)?
```suggestion
    sourceFile = file(Paths.get("$rootDir" + zipFileName))
```
Oder geht es so?
```suggestion
    sourceFile = file(Paths.get(${rootDir} + zipFileName))
```
Ich habe manchmal auch Mühe, wo man nun welche Schreibweise benutzen kann.

---

> Ermittlung, wie der s3TargetBucket je nach Umgebung heisst, auf eine Zeile vereinfacht. Das muss man nicht genau verstehen???

Nein :-)
Doku wäre aber hier: http://groovy-lang.org/operators.html#_ternary_operator
Mit dem Ternary Operator wird die Variable abhängig davon, ob `(gretlEnvironment == 'production')` true oder false ist, mit dem Wert hinter dem Fragezeichen oder demjenigen hinter dem Doppelpunkt initialisiert.

---

sourceFile = file(Paths.get(${rootDir} + zipFileName)) geht auch nicht, ich lasse es mal so.
sourceFile = file(Paths.get("$rootDir" + zipFileName))
Das wegen dem Ternary Operator war eher ironisch gemeint. Aber wie würde man das mit mehr als 3 Optionen machen?

---

> sourceFile = file(Paths.get(${rootDir} + zipFileName)) geht auch nicht, ich lasse es mal so.
Ah Mist, ich meinte:
`sourceFile = file(Paths.get("$rootDir", zipFileName))` (mit Komma statt Plus)
oder
`sourceFile = file(Paths.get(${rootDir}, zipFileName))` (mit Komma statt Plus)

Aber egal, es ist nicht so wichtig.

---

> Aber wie würde man das mit mehr als 3 Optionen machen?

Es geht eigentich gar nicht um 3 oder mehr oder weniger Optionen, sondern es ist immer noch ein einfaches if - else, aber anders geschrieben. Es ist gewissermassen folgendes auf eine Zeile gepackt:
```
if (gretlEnvironment == 'production') {
    s3TargetBucket = 'ch.so.agi.geodata'
} else {
    s3TargetBucket = "ch.so.agi.geodata-$gretlEnvironment"
}
```
D.h. es gibt nur einen speziellen Fall, nämlich production. Bei den anderen wird einfach $gretlEnvironment angehängt. So wäre es sogar ready für noch mehr Umgebungen.